### PR TITLE
Update ChangeLog for 5.1.0.

### PR DESCRIPTION
**Pull request by interwq**

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -4,6 +4,117 @@ brevity.  Much more detail can be found in the git revision history:
...
+  - Add maximum background thread count option.  (@djwatson)
+  - Allow prof_active to control opt.lg_prof_interval and prof.gdump.
+    (@interwq)
+  - Allow arena index lookup based on allocation addresses via mallctl.
+    (@lionkov)
+  - Allow disabling initial-exec TLS model.  (@davidtgoldblatt, @KenMacD)
+  - Add opt.lg_extent_max_active_fit to set the max ratio between the size of
+    the active extent selected (to split off from) and the size of the requested
+    allocation.  (@interwq, @davidtgoldblatt)
+  - Add retain_grow_limit to set the max size when grow virtual address space.
```
Comment (jasone) [inline ChangeLog]: s/grow/growing/ ?

File: ChangeLog [hunk 2]
```diff
@@ -4,6 +4,117 @@ brevity.  Much more detail can be found in the git revision history:
...
+  - Make sure CXXFLAGS is tested with CPP compiler.  (@nehaljwani)
+  - Fix 32-bit build on MSVC.  (@rustyx)
+  - Fix external symbol on MSVC.  (@maksqwe)
+  - Avoid a printf format specifier warning.  (@jasone)
+  - Add configure option --disable-initial-exec-tls which can allow jemalloc to
+    be dynamically loaded after program startup.  (@davidtgoldblatt, @KenMacD)
+  - AArch64: Add ILP32 support.  (@cmuellner)
+
+  Optimizations and refactors:
+  - Improve active extent fit with extent_max_active_fit. This considerably
```
Comment (jasone) [inline ChangeLog]: s/\. /.  /

File: ChangeLog [hunk 3]
```diff
@@ -4,6 +4,117 @@ brevity.  Much more detail can be found in the git revision history:
...
+  - Refactor OOM handling in newImpl.  (@wqfish)
+  - Add internal fine-grained logging functionality for debugging use.
+    (@davidtgoldblatt)
+  - Refactor arena / tcache interactions.  (@davidtgoldblatt)
+  - Refactor extent management with dumpable flag.  (@davidtgoldblatt)
+  - Add runtime detection of lazy purging.  (@interwq)
+  - Use pairing heap instead of red-black tree for extents_avail.  (@djwatson)
+  - Use sysctl on startup in FreeBSD.  (@trasz)
+  - Use thread local prng state instead of atomic.  (@djwatson)
+  - Decay optimization by over purge 1 large extent.  (@interwq)
```
Comment (jasone) [inline ChangeLog]: ?

File: ChangeLog (hunk 1 above)
Comment (interwq) [reply ChangeLog]: will fix

File: ChangeLog (hunk 2 above)
Comment (interwq) [reply ChangeLog]: Will fix (double space).

File: ChangeLog (hunk 3 above)
Comment (interwq) [reply ChangeLog]: Ah yeah will try explaining this carefully.

File: ChangeLog [hunk 4]
```diff
@@ -4,6 +4,120 @@ brevity.  Much more detail can be found in the git revision history:
...
+  - Refactor arena / tcache interactions.  (@davidtgoldblatt)
+  - Refactor extent management with dumpable flag.  (@davidtgoldblatt)
+  - Add runtime detection of lazy purging.  (@interwq)
+  - Use pairing heap instead of red-black tree for extents_avail.  (@djwatson)
+  - Use sysctl on startup in FreeBSD.  (@trasz)
+  - Use thread local prng state instead of atomic.  (@djwatson)
+  - Make decay to always purge one more extent than before, because in
+    practice large extents are usually the ones that cross the decay threshold.
+    Purging the additional extent helps save memory as well as reduce VM
+    fragmentation.  (@interwq)
```
Comment (interwq) [inline ChangeLog]: @jasone : does this look reasonable to you? More details in https://github.com/jemalloc/jemalloc/pull/1092.

**Pull request merged**